packages/cli/scripts/pre-designed-nft.ts

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+import fs from 'fs'
+import path from 'path'
+import { Configuration, CreateImageRequestSizeEnum, OpenAIApi } from 'openai'
+import { create as ipfsHttpClient } from 'ipfs-http-client'
+import { validateTokenBaseUriForPreDesignedCollection, NFTMetadata } from '@alephium/web3'
+import { parse, stringify } from 'yaml'
+
+export interface NFTMetadataConfig {
+  name?: string
+  description?: string
+  attributes?: [
+    {
+      [key: string]: string | number | boolean
+    }
+  ]
+}
+
+const validMetadataFields = ['name', 'description', 'attributes']
+const validMetadataAttributeTypes = ['string', 'number', 'boolean']
+
+export async function generateImagesWithOpenAI(
+  openaiApiKey: string,
+  prompt: string,
+  numberOfImages: number,
+  imageSize: CreateImageRequestSizeEnum,
+  storedDir: string
+) {
+  if (!fs.existsSync(storedDir)) {
```

Comment:
How about creating the directory if it does not exist?

Reply:
Fixed, thanks :+1: